update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,341 @@
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        GatewayResponse<Problem> response =  sendRequest(updateDoiRequest, publication.getOwner());
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_BAD_REQUEST)));
+
+        assertThat(details.getDetail(),
+            containsString("You must initiate creation of a DoiRequest before you can update it."));
+    }
+
+    @Test
+    public void handleRequestReturnsForbiddenExceptionWhenInputUsernameIsNotThePublicationOwner()
+            throws IOException {
+        final TestAppender appender = LogUtils.getTestingAppender(DynamoDBDoiRequestsService.class);
+        Publication publication = insertPublicationWithoutDoiRequest(
+            Clock.fixed(mockOneHourBefore, ZoneId.systemDefault()));
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, INVALID_USERNAME);
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_FORBIDDEN)));
+
+        assertThatProblemDetailsDoesNotRevealSensitiveInformation(details, INVALID_USERNAME,
+                validUsername(publication));
+
+        assertThatLogsContainReasonForForbiddenMessage(appender, publication);
+    }
+
+    @Test
+    public void handleRequestReturnsForbiddenExceptionWhenUsernameIsNotInRequestContext()
+        throws IOException {
+        final TestAppender appender = LogUtils.getTestingAppender(UpdateDoiRequestHandler.class);
+        Publication publication = insertPublicationWithoutDoiRequest(
+            Clock.fixed(mockOneHourBefore, ZoneId.systemDefault()));
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        var requestContext = objectMapper.createObjectNode();
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, requestContext);
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_FORBIDDEN)));
+
+        assertThat(details.getDetail(), is(equalTo("Forbidden")));
+        assertThat(appender.getMessages(),
+            containsString("Missing from requestContext: /authorizer/claims/custom:feideId"));
+    }
+
+    private void assertThatProblemDetailsDoesNotRevealSensitiveInformation(Problem details,
```

Comment:
Re-arrange your methods so that the private methods are below the public methods (Tests). Ctrl-shift-w